packages/backend/src/tokens/request.ts

```diff
@@ -531,3 +631,106 @@ export const debugRequestState = (params: RequestState) => {
   const { isSignedIn, proxyUrl, reason, message, publishableKey, isSatellite, domain } = params;
   return { isSignedIn, proxyUrl, reason, message, publishableKey, isSatellite, domain };
 };
+
+/**
+ * Determines if the given URL and settings indicate a desire to activate a specific
+ * organization or personal workspace.
+ *
+ * @returns {OrganizationSyncTarget | null} - The function can return the following:
+ *  - {OrganizationSyncTarget}: If the URL and settings indicate a desire to activate an organization or personal workspace.
+ */
+export function getOrganizationSyncTarget(
+  url: URL,
+  options: OrganizationSyncOptions | undefined,
+): OrganizationSyncTarget | null {
+  if (!options) {
+    return null;
+  }
+
+  // Check for personal workspace activation
+  if (options.personalWorkspacePatterns) {
+    let matcher: MatchFunction<Partial<Record<string, string | string[]>>>;
+    try {
+      matcher = match(options.personalWorkspacePatterns);
+    } catch (e) {
+      // Likely to be encountered during development, so throwing the error is more prudent than logging
+      throw new Error(`Invalid personal workspace pattern "${options.personalWorkspacePatterns}": "${e}"`);
```

Comment:
Note that in practice, this never seems to throw - any string appears to be accepted.